pkg/ir/spec.go

```diff
@@ -53,36 +57,151 @@ type (
 
 	// ConfigDefs holds definitions that are part of the network architecture
 	ConfigDefs struct {
-		Subnets map[string]IP
+		VPCs map[ID]*VPCDetails
 
-		// Network interface name to IP address
-		NIFToIP map[string]IP
+		Subnets map[ID]*SubnetDetails
 
-		// Instance is a collection of NIFs
-		InstanceToNIFs map[string][]string
+		NIFs map[ID]*NifDetails
 
-		// VPEs have a single IP
-		VPEToIP map[string]IP
+		Instances map[ID]*InstanceDetails
 
-		// list of VPC's cidrs
-		AddressPrefixes []CIDR
+		VPEEndpoints map[ID]*VPEEndpointDetails
+
+		VPEs map[ID]*VPEDetails
 	}
 
 	// Definitions adds to ConfigDefs the spec-specific definitions
 	Definitions struct {
 		ConfigDefs
 
 		// Segments are a way for users to create aggregations.
-		SubnetSegments map[string][]string
+		SubnetSegments map[ID]*SubnetSegmentDetails
 
-		// key = name of the segment. value = map where its key is the cidr and its value is the contained subnets
-		CidrSegments map[string]map[string][]string
+		// Cidr segment might contain some cidrs
+		CidrSegments map[ID]*CidrSegmentDetails
 
 		// Externals are a way for users to name IP addresses or ranges external to the VPC.
-		Externals map[string]IP
+		Externals map[ID]*ExternalDetails
+	}
+
+	VPCDetails struct {
+		AddressPrefixes []CIDR
+		// tg
+		// lb
+	}
+
+	SubnetDetails struct {
+		NamedEntity
+		CIDR IP
```

Comment:
Why not `CIDR CIDR`?

Reply:
From the beginning there was the confusion between `IP` and `CIDR` in the code (mainly that subnets contained a field of type `IP` and not `CIDR`). At first I tried to solve the problem, but after further thought I decided to leave it as it is for now, since the next thing I'm going to do is to convert the types `IP` and `CIDR` to `IPBlock`.
Would you like me to fix the problem? Or I can go straight to (#121).

Reply:
This can wait for #121